airflow/executors/local_executor.py

```diff
@@ -201,161 +124,27 @@ class LocalExecutor(BaseExecutor):
-            self.queue: Optional['Queue[ExecutorWorkType]'] = None
-
-        def start(self) -> None:
-            """Starts limited parallelism implementation."""
-            if not self.executor.manager:
-                raise AirflowException(NOT_STARTED_MESSAGE)
-            self.queue = self.executor.manager.Queue()
-            if not self.executor.result_queue:
-                raise AirflowException(NOT_STARTED_MESSAGE)
-            self.executor.workers = [
-                QueuedLocalWorker(self.queue, self.executor.result_queue)
-                for _ in range(self.executor.parallelism)
-            ]
-
-            self.executor.workers_used = len(self.executor.workers)
-
-            for worker in self.executor.workers:
-                worker.start()
-
-        def execute_async(
-            self,
-            key: TaskInstanceKey,
-            command: CommandType,
-            queue: Optional[str] = None,
-            executor_config: Optional[Any] = None,
-        ) -> None:
-            """
-            Executes task asynchronously.
-
-            :param key: the key to identify the task instance
-            :param command: the command to execute
-            :param queue: name of the queue
-            :param executor_config: configuration for the executor
-            """
-            if not self.queue:
-                raise AirflowException(NOT_STARTED_MESSAGE)
-            self.queue.put((key, command))
-
-        def sync(self):
-            """Sync will get called periodically by the heartbeat method."""
-            while True:
-                try:
-                    results = self.executor.result_queue.get_nowait()
-                    try:
-                        self.executor.change_state(*results)
-                    finally:
-                        self.executor.result_queue.task_done()
-                except Empty:
-                    break
-
-        def end(self):
-            """Ends the executor. Sends the poison pill to all workers."""
-            for _ in self.executor.workers:
-                self.queue.put((None, None))
-
-            # Wait for commands to finish
-            self.queue.join()
-            self.executor.sync()
+        self.futures = {}
 
     def start(self) -> None:
         """Starts the executor"""
         old_proctitle = getproctitle()
         setproctitle("airflow executor -- LocalExecutor")
-        self.manager = Manager()
         setproctitle(old_proctitle)
-        self.result_queue = self.manager.Queue()
-        self.workers = []
-        self.workers_used = 0
-        self.workers_active = 0
-        self.impl = (
-            LocalExecutor.UnlimitedParallelism(self)
-            if self.parallelism == 0
-            else LocalExecutor.LimitedParallelism(self)
-        )
+        # This isn't _truely_ unlimited, but it should be good enough!
+        size = 2**30 if self.parallelism == 0 else self.parallelism
```

Comment:
Interestingly there is no promises from the unlimited: https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#parallelism

Reply:
Huh. TIL

Sadly I think that although this isn't documented it's still "relied upon" behaviour so we can't just remove it.